widget/embedded/src/store/slices/data.ts

```diff
@@ -1,17 +1,80 @@
 // We keep all the received data from server in this slice
 
 import type { ConfigSlice } from './config';
-import type { Balance, TokenHash } from '../../types';
+import type { CachedEntries } from '../../services/cacheService';
+import type {
+  Balance,
+  Blockchain,
+  BlockchainAndTokenConfig,
+  TokenHash,
+} from '../../types';
 import type { Asset, BlockchainMeta, SwapperMeta, Token } from 'rango-sdk';
 import type { StateCreator } from 'zustand';
 
+import { cacheService } from '../../services/cacheService';
 import { httpService as sdk } from '../../services/httpService';
 import { compareWithSearchFor, containsText } from '../../utils/common';
-import { isTokenExcludedInConfig } from '../../utils/configs';
 import { createTokenHash, isTokenNative } from '../../utils/meta';
 import { sortLiquiditySourcesByGroupTitle } from '../../utils/settings';
 import { areTokensEqual, compareTokenBalance } from '../../utils/wallets';
 
+export const calculateSupportedTokens = (params: {
```

Comment:
To be honest, this function is seems to be written in Japanese to me , Could you clear the assumptions by adding some comments on the flow?

Reply:
Also you can create a `store/utils/data.ts` to move this function to that file.

Reply:
Please add some unit tests to test and show how it should behave more clearly. You can create a `store/utils/data.test.ts` to put your unit tests.